optimum/exporters/features.py

```diff
@@ -0,0 +1,744 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Model export features manager."""
```

Comment:
I always found the word "features" a bit confusing in the original exporter - does it refer to feature vectors or something else? 

Since we have the opportunity to do something new, would something like "architecture" or "head" be more appropriate?

Reply:
👍 for architecture, or what about just `pipeline type` or `task type`?

More generally 100% agree that we could use this opportunity to improve the DX as much as we can

Reply:
@osanseviero

Reply:
I like:

- `TaskManager`
- `ArchitectureManager`

Reply:
either `task` or `pipeline` would be consistent with the rest of the Hub. If consistent with the Hub, then `task` fits better. If consistent with `transformers`, then `pipeline`.

Reply:
I went for task!
Actually I think that pipeline is not suitted because we do not match the pipeline names. Like `sequence-classification` instead of `text-classification`.

Reply:
Task sounds good (even if we can't 100% match the Hub definitions because of past key-value variants + the fact we're really mapping "task" to `AutoModelForXxx`).